### PR TITLE
Make SPI local properties json serializable

### PR DESCRIPTION
**Pull request by erichwang**

### CODE REVIEW

File: presto-spi/src/main/java/com/facebook/presto/spi/SortingProperty.java [hunk 1]
```diff
@@ -48,6 +54,7 @@ public E getColumn()
         return Collections.singleton(column);
     }
 
+    @JsonProperty
     public SortOrder getOrder()
```
Comment (electrum) [inline presto-spi/src/main/java/com/facebook/presto/spi/SortingProperty.java]: Are you fine with the default serialization for this enum?

File: presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestLocalProperties.java [hunk 2]
```diff
@@ -703,6 +713,34 @@ public void testDifferentSortOrders()
                 Optional.of(sorted("a", SortOrder.ASC_NULLS_LAST)));
     }
 
+    @Test
+    public void testJsonSerialization()
+            throws Exception
+    {
+        ObjectMapper mapper = new ObjectMapperProvider().get()
+                .registerModule(new SimpleModule()
+                        .addDeserializer(ColumnHandle.class, new JsonDeserializer<ColumnHandle>()
```
Comment (electrum) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestLocalProperties.java]: This is duplicated in `TestDupleDomain`. Create a `TestingColumnHandleDeserializer`?

File: presto-spi/src/main/java/com/facebook/presto/spi/SortingProperty.java (hunk 1 above)
Comment (erichwang) [inline presto-spi/src/main/java/com/facebook/presto/spi/SortingProperty.java]: yes

File: presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestLocalProperties.java (hunk 2 above)
Comment (erichwang) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestLocalProperties.java]: the two packages don't have a dependency on each other yet. this isn't important enough to make them dependent.

**Pull request closed without merging**